File: Daily Challanges/2026-03-March/2026-03-29.py
def is_valid_isbn10(s):
    s = s.replace("-", "")
    if len(s) != 10:
        return False
    total = 0
    for i in range(10):
        if i < 9:
            if not s[i].isdigit():
                return False
            value = int(s[i])
        else:
            if s[i] == "X":
                value = 10
            elif s[i].isdigit():
                value = int(s[i])
            else:
                return False
        total += value * (i + 1)
    return total % 11 == 0

File: Daily Challanges/2026-03-March/2026-03-29.js
function isValidIsbn10(s) {
  s = s.replace(/-/g, "");
  if (s.length !== 10) return false;
  let total = 0;
  for (let i = 0; i < 10; i++) {
    let value;
    if (i < 9) {
      if (!/^\d$/.test(s[i])) return false;
      value = Number(s[i]);
    } else {
      if (s[i] === "X") {
        value = 10;
      } else if (/^\d$/.test(s[i])) {
        value = Number(s[i]);
      } else {
        return false;
      }
    }
    total += value * (i + 1);
  }
  return total % 11 === 0;
}
